src/interval_heap.rs

```diff
@@ -338,6 +351,21 @@ impl<T, C: Compare<T>> IntervalHeap<T, C> {
     pub fn clear(&mut self) {
         self.data.clear();
     }
+
+    fn is_valid(&self) -> bool {
```

Comment:
Now that this isn't just "test" code, I'd appreciate some documentation of this.